### DISCUSSION TITLE
Set recording settings

### DISCUSSION THREAD

**Discussion by jpbast** [Palakis/obs-websocket#1013] [Q&A]
Is there any way to set the following recording settings?
![image](https://user-images.githubusercontent.com/39203590/190639561-4d6dad7d-2725-4bba-9e10-09b3090fb0df.png)
I managed to set some options with `SetVideoSettings` but not everything. I tried to use `SetOutputSettings` either but didn't figure out what options are available in `outputSettings`.

**Comment by heartz66** [comment]
This might help you: https://github.com/obs-websocket-community-projects/obs-websocket-js/blob/fdca73c46e1c4f5e5decabe5760481077c035732/src/types.ts#L898

**Reply by jpbast** [reply]
With these options I can change video quality but I still can't figure out how to set the recording path and the recording format

**Reply by courgeon** [reply]
Agreed ... Seems like the 4.x method SetFilenameFormatting is gone. That's a dealbreaker for any advanced usage. Switching back to OBS 27 and OWS 4.x. Too bad.

**Comment by slaudel** [comment]
Is there any progress on that topic?
We too used to set recording file names and directories with websocket 4.9 but are struggeling to do so with the new protocol.

**Comment by MMmaomao** [comment]
SetRecordDirectory is work fine, can anyone know how to set the recording format? tks
